ext/opcache/jit/zend_jit_helpers.c

```diff
@@ -779,7 +779,13 @@ static zval* ZEND_FASTCALL zend_jit_fetch_dim_w_helper(zend_array *ht, zval *dim
 			}
 		}
 	} else {
-		retval = zend_hash_add_new(ht, offset_key, &EG(uninitialized_zval));
+		zend_ulong idx;
+
+		if (ZEND_HANDLE_NUMERIC(offset_key, idx)) {
```

Comment:
This should be handled higher up, directly below the `str_index` label. Otherwise, the above lookup may fail to find an existing element.

It also looks like this code is duplicated quite a few times, so we should fix the other functions above in the same way.

Reply:
OK, let me try a slightly different way of fixing this function...

Reply:
OK, please see if it is better now.